### PR TITLE
GCM to FCM migration in PushPayloadHelper

### PR DESCRIPTION
**Pull request by wkal-pubnub**

### CODE REVIEW

File: pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt [hunk 1]
```diff
@@ -127,25 +126,253 @@ class PushPayloadHelper {
...
+                return buildMap {
+                    collapseKey?.let { put("collapse_key", it) }
+                    put("priority", priority.name)
+                    ttl?.let { put("ttl", it) }
+                    restrictedPackageName?.let { put("restricted_package_name", it) }
+                    data?.let { put("data", it) }
+                    notification?.let { put("notification", it.toMap()) }
+                    fcmOptions?.let { put("fcm_options", it.toMap()) }
+                    put("direct_boot_ok", directBootOk)
+                }
```
Comment (marcin-cebo) [inline pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt:375]: Don't we by any chance need to add "click_action" to Notification to handle following situation:

{
  "message": {
    "topic": "news",
    "notification": {
      "title": "Breaking News",
      "body": "New news story available."
    },
    "data": {
      "story_id": "story_12345"
    },
    "android": {
      "notification": {
        "click_action": "TOP_STORY_ACTIVITY",
        "body": "Check out the Top Story"
      }
    },
    "apns": {
      "payload": {
        "aps": {
          "category" : "NEW_MESSAGE_CATEGORY"
        }
      }
    }
  }
}

File: pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt [hunk 2]
```diff
@@ -127,25 +126,253 @@ class PushPayloadHelper {
...
+        }
+
+        class ApnsConfig : PushPayloadSerializer {
+            var headers: Map<String, String>? = null
+            var payload: Map<String, String>? = null
+            var fcmOptions: ApnsFcmOptions? = null
+
+            class ApnsFcmOptions : PushPayloadSerializer {
+                var analyticsLabel: String? = null
+                var image: String? = null
```
Comment (marcin-cebo) [inline pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt:216]: In this page there is none "image". How do you know that it is required?

Comment (wkal-pubnub) [reply pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt:216]: you mean in https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#apnsfcmoptions ? I see an optional image there

File: pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt (hunk 1 above)
Comment (wkal-pubnub) [reply pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt:375]: ha, you're right, I missed one! It's in AndroidNotification though. added

File: pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt (hunk 2 above)
Comment (marcin-cebo) [reply pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt:216]: Thanks for clarification.

**Review by marcin-cebo — APPROVED** [review]

**Pull request merged**